cardano-cli/src/Cardano/CLI/Types/Errors/TxValidationError.hs

```diff
@@ -272,57 +212,33 @@ validateTxCertificates era certsAndScriptWitnesses = cardanoEraConstraints era $
         Just sWit -> (sCred, ScriptWitness ScriptWitnessForStakeAddr sWit)
         Nothing   -> (sCred, KeyWitness KeyWitnessForStakeAddr)
 
-newtype TxProtocolParametersValidationError
-  = ProtocolParametersNotSupported AnyCardanoEra
-  deriving Show
-
-instance Error TxProtocolParametersValidationError where
-  prettyError (ProtocolParametersNotSupported e) =
-    "Transaction protocol parameters are not supported in " <> pretty e
-
 validateProtocolParameters
   :: CardanoEra era
```

Comment:
This will only fail if we call `validateProtocolParameters` with `ByronEra`. However the code won't compile if we call this with `ByronEra`:

```
ghci> validateProtocolParameters ByronEra Nothing

<interactive>:22:1: error: [GHC-39999]
    • No instance for ‘IsShelleyBasedEra ByronEra’
        arising from a use of ‘print’
    • In a stmt of an interactive GHCi command: print it
```

What we should do is change the parameter from `CardanoEra era` to `ShelleyBasedEra era` but actually in this case no validation is done! 

So we can remove this function and directly use `setTxProtocolParams`.

Have a look at the other functions that demand `CardanoEra era` but should be parameterized on `ShelleyBasedEra era`. If no validation is done, we can just drop them. I'd do that in a separate PR.

Reply:
OK will do in a follow-up PR right away :+1: 

@Jimbo4350> so can you approve in the meantime?

Reply:
I can't because it would be adding churn. If you do what I explained above it should also result in the removal of the unnecessary error constructors.